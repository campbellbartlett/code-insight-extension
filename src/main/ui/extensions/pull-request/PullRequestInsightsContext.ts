export type CodeInsightReport = {
    name: string;
    status: string;
};

export type PullRequestInsightsContext = {
    codeInsightReports: [CodeInsightReport];
    commitHash: string;
    repositorySlug: string;
    projectKey: string;
    userAdmin: boolean;
    adminOverride: boolean;
};
