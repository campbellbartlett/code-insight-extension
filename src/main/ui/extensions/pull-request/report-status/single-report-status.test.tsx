import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import SingleReportStatus from './single-report-status';
import { CodeInsightReport } from '../PullRequestInsightsContext';
import PassedStatus from './passed-status';
import WaitingStatus from './waiting-status';
import FailedStatus from './failed-status';

const testReportTitle = 'Test Report';

const passedReport: CodeInsightReport = {
    name: testReportTitle,
    status: 'PASS',
};

const waitingReport: CodeInsightReport = {
    name: testReportTitle,
    status: 'WAITING',
};

const failedReport: CodeInsightReport = {
    name: testReportTitle,
    status: 'FAIL',
};

describe('single report status shows correct data', () => {
    it('should render passed status correctly', () => {
        const notePreview = shallow(<SingleReportStatus report={passedReport} />);

        expect(notePreview).not.toBeNull();
        expect(notePreview.contains(<PassedStatus />)).toBeTruthy();
        expect(notePreview.find('#report-name').text()).toEqual(testReportTitle);
    });

    it('should render waiting status correctly', () => {
        const notePreview = shallow(<SingleReportStatus report={waitingReport} />);

        expect(notePreview).not.toBeNull();
        expect(notePreview.contains(<WaitingStatus />)).toBeTruthy();
        expect(notePreview.find('#report-name').text()).toEqual(testReportTitle);
    });

    it('should render failed status correctly', () => {
        const notePreview = shallow(<SingleReportStatus report={failedReport} />);

        expect(notePreview).not.toBeNull();
        expect(notePreview.contains(<FailedStatus />)).toBeTruthy();
        expect(notePreview.find('#report-name').text()).toEqual(testReportTitle);
    });

    it('should match snapshot', () => {
        const notePreview = shallow(<SingleReportStatus report={passedReport} />);

        expect(notePreview).not.toBeNull();
        expect(shallowToJson(notePreview)).toMatchSnapshot();
    });
});
