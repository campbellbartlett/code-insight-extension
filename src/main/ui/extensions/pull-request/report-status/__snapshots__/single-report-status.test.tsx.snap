// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single report status shows correct data should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "width": "10rem",
    }
  }
>
  <span
    id="report-name"
  >
    Test Report
  </span>
   
  <PassedStatus />
</div>
`;
